### ISSUE TITLE
[Storage & Angular6] Error on put method with aws-amplify 

### ISSUE THREAD

**Issue by Loghrenar** [aws-amplify/amplify-js#1109]
Do you want to request a feature or report a bug?

In my opinion it's a **bug**

What is the current behavior?

I was working with latest version of **Angular5** with aws-amplify and **Storage** service to upload files in a specific **S3** bucket (which was working). For reasons i switched to **Angular6** and it seems like i'm facing errors i never saw before.

=> `StorageClass - ensure credentials error: TypeError: Cannot read property 'Stream' of undefined
    at Object.computeSha256`

This is my current amplify configuration (environment.ts file):

```     amplify: {
        Auth: {
            identityPoolId: 'xxxxxxxx',
            region: 'xxxxxx',
            userPoolId: 'xxxxxx',
            userPoolWebClientId: 'xxxxxxxx',
            mandatorySignIn: false,
            cookieStorage: {
                domain: 'localhost',
                path: '/',
                expires: 365,
                secure: false
            }
        },
        Storage: {
            bucket: 'arn:aws:s3:::example-name',
            region: 'xxxxxxxxx',
        }
    },
```
This is my service:

```
import { Injectable } from '@angular/core';
import { Observable, from, BehaviorSubject, of } from 'rxjs';
import { map, tap, catchError } from 'rxjs/operators';
import Amplify, { Storage } from 'aws-amplify';
import { environment } from './../../environments/environment';

@Injectable({ providedIn: 'root' })
export class AuthService {

    constructor() {
        Amplify.configure(environment.amplify);
    }

    updateFile(file: string, option: string, object?: object): Observable<any> {
        return from(Storage.put(file, option, object));
    }

    removeFile(file: string): Observable<any> {
        return from(Storage.remove(file));
    }
}
```

Example of use:

```
        this.storageService.updateFile(fileName, file, {
            level: 'public',
            contentType: file.type,
        })
```

I have the basic buckets policies given in the tutorial on the documentation

Which versions of Amplify, and which browser / OS are affected by this issue? Did this work in previous versions?

Current versions:
Angular 6.0.8
Amplify 0.4.6
Mac OS High Sierra 10.13.3
Chrome 67.0.3396.87

This was working with latest Angular5 version and same versions for others libraries/os/browser

I may be wrong on some points so i'm open to any feedback, thanks in advance.

**Comment by Antonyselt** [comment]
I have the same issue here! It works just fine in angular 4, I follow a tutorial and it works but I try to duplicate the example with angular 6 and I always getting this Error: ` Cannot read property 'Stream' of undefined`

**Comment by Loghrenar** [comment]
Hey @powerful23, do you have any news about the investigations on this problem ?

Thanks in advance.

**Comment by haverchuck** [comment]
@Loghrenar 

Thanks so much for reaching out about this.  Our documentation has some notes about Angular 6.  Have you taken the steps noted [here](https://aws.github.io/aws-amplify/media/angular_guide#angular-6-support)?   Please let us know!

Thanks again.

**Comment by Loghrenar** [comment]
Hey @haverchuck,

Thank you for your feedback.

Yes i already read the documentation but i'm currently using the "classic" version of amplify not the angular specific one As I said previously, this bug didn't happen in Angular5

**Comment by haverchuck** [comment]
Hi @Antonyselt 

Would you mind sharing which tutorial you are following? 

Thanks!

**Comment by Loghrenar** [comment]
Hey @haverchuck, 

I know you're asking @Antonyselt, for my part it's this one:
https://aws.github.io/aws-amplify/media/storage_guide.html

I think we followed the same.

**Comment by Antonyselt** [comment]
I follow two tutorials but none with aws-amplify but the error is the same

https://medium.com/@bevin.hernandez/angular-5-file-uploads-to-s3-with-relative-ease-ea19b7fd120e

http://javasampleapproach.com/frontend/angular/angular-4-amazon-s3-example-how-to-upload-file-to-s3-bucket

Both works in angular 4 and 5 but not in angular 6

**Comment by Loghrenar** [comment]
Hey @haverchuck,

For my part, i upgraded an existing project from Angular 5 to 6.
Which version of amplify were you using in this test ? My version is 0.4.6 maybe if you're using the latest and it's working ?

yes, "node" is in my types array.

**Comment by Antonyselt** [comment]
I started a new project with angular 6, and yes I have pute node in mi types.

Maybe could you give us a template or maybe can we se your code. to see whats wrong with ours

**Comment by Loghrenar** [comment]
Hey,
Do we have some news about this problem ?
Thanks.

**Comment by medbz** [comment]
Okay , I have faced the same error before in my angular 6 application . Just you have to add this code in your index.html `  <script>
    var global = global || window;
    var Buffer = Buffer || [];

  </script>`

**Comment by Loghrenar** [comment]
A little update:
- Same error: 
```
{[WARN] 51:25.704 StorageClass - ensure credentials error: TypeError: Cannot read property 'Stream' of undefined
    at Object.computeSha256 (http://localhost:…}
```
- New error:
```
ERROR Error: Uncaught (in promise): No credentials
    at resolvePromise (zone.js:814)
    at resolvePromise (zone.js:771)
    at zone.js:724
    at step (Storage.js:18)
    at fulfilled (Storage.js:16)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:388)
    at Object.onInvoke (core.js:3820)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:387)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:138)
    at zone.js:872
```

I tried to make it works a new project under Angular7 & amlify `^1.1.7`, it works on this new project with the fix of @medbz. So it seems like something else is blocking.

I'm investigating, i will update this issue if I can fix it.

**Comment by Loghrenar** [comment]
So, as i expected i'm using `angular-instantsearch`, adding the package and the fix 
```
(window as any).process = {
  env: { DEBUG: undefined },
};
```
in the `polyfills.ts` throw the error 
```
StorageClass - ensure credentials error: TypeError: Cannot read property 'Stream' of undefined
```
with error message `No credentials`.

(full logs with credentials problem)
```
[DEBUG] 28:14.135 AuthClass - Getting current user credentials
ConsoleLogger.js:100 {[DEBUG] 28:14.138 AuthClass - failed to get or parse item aws-amplify-federatedInfo: SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at AuthCla…}
ConsoleLogger.js:84 [DEBUG] 28:14.141 AuthClass - Getting current session
ConsoleLogger.js:100 {[DEBUG] 28:14.212 AuthClass - Getting the session from this user:: CognitoUser}
ConsoleLogger.js:100 {[DEBUG] 28:14.215 AuthClass - Succeed to get the user session: CognitoUserSession}
ConsoleLogger.js:100 {[DEBUG] 28:14.224 AuthClass - getting session success: CognitoUserSession}[DEBUG] 28:14.224 AuthClass - getting session success: CognitoUserSession {idToken: CognitoIdToken, refreshToken: CognitoRefreshToken, accessToken: CognitoAccessToken, clockDrift: 1}__proto__: Object
ConsoleLogger.js:84 [DEBUG] 28:14.227 Credentials - set credentials from session
ConsoleLogger.js:100 {[DEBUG] 28:14.230 AuthClass - getting session failed: TypeError: Cannot read property 'Stream' of undefined
    at Object.computeSha256 (http://localhost:…}[DEBUG] 28:14.230 AuthClass - getting session failed: TypeError: Cannot read property 'Stream' of undefined
    at Object.computeSha256 (http://localhost:4201/vendor.js:181204:32)
    at Request.COMPUTE_SHA256 (http://localhost:4201/vendor.js:172609:18)
    at Request.callListeners (http://localhost:4201/vendor.js:177453:18)
    at Request.emit (http://localhost:4201/vendor.js:177429:10)
    at Request.emit (http://localhost:4201/vendor.js:176063:14)
    at Request.transition (http://localhost:4201/vendor.js:175402:10)
    at AcceptorStateMachine.runTo (http://localhost:4201/vendor.js:180457:12)
    at http://localhost:4201/vendor.js:180469:10
    at Request.<anonymous> (http://localhost:4201/vendor.js:175418:9)
    at Request.<anonymous> (http://localhost:4201/vendor.js:176065:12)__proto__: Object
ConsoleLogger.js:84 [DEBUG] 28:14.231 Credentials - setting credentials for guest
```

Seems like 
```
   var global = global || window;
   var Buffer = Buffer || [];
```
in `index.html` is confliction with the algolia modifications in the `polyfills.ts`.
 
So, i don't know if this bug still belongs here or should be reported on `angular-instantsearch` repository. Waiting for an answer to let this issue opened or close it.

**Comment by jordanranz** [comment]
Hey @Loghrenar, are you still experiencing this issue with the latest version of Amplify?

Did you get any useful information related to `angular-instantsearch`?

**Comment by Loghrenar** [comment]
Hey @jordanranz,

I opened an issue in the Algolia repository with no success, I just removed `angular-instantsearch` because we didn't need it anymore (business changes). Now, obviously, Amplify works well and I don't face any problems anymore. 

I'll check the next  version of `angular-instantsearch` in the future to see if there are still conflicts. This is issue can be definitely closed. Thank you.
